transcom/mymove
# B 21583 int

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1025118)

## Summary

When a move is created that has an Alaska duty location, we need to make it so it routes to the right GBLOC for that zone.
Needs to route through SC, TOO, TIO and show in the correct GBLOC for each queue.

### How to test

1. Login as a customer, create a new move for each Zone and scenario below.  Six moves total.  Zone tips are below.  There is some sql below too to help make sure your city/zip is pulling back the correct GBLOC you expect.
2. Be sure to pick an AK origin Duty Location and AK Pickup Address.
3. Add HHG shipment, Submit move as customer, you can check the move in the database to verify the orders table has the correct GBLOC at this point.
4. Login as the SC for each GBLOC to verify that the moves show in the queue as expected.
5. Do a search under Move Search tab, make sure the Origin GBLOC is as expected in the results.
6. Before submitting as SC, also login as the HQ user and verify you see the moves in the queue there as well.
7. Submit your moves and then login as TOO/HQ for each move and verify you see them in the right queue/GBLOC for the pickup address you chose.
8. Approve your moves, process the moves in Prime Sim so that you can get them ready for TIO user.
9. Login as TIO users and verify you can see all moves in the queues as expected.
10.  Login as an SC, create a new move for each Zone scenario, add a shipment.  
11. Search for each new SC created move in the move search tab and verify that the GBLOC listed in the search results is what you expect from the scenarios below.

**Zones**:
- Zone I shipments will be processed by JPPSO Elmendorf-Richardson (GBLOC MBFL).
- Zone II shipments are divided between JPPSO-NW (GBLOC JEAT) and JPPSO Elmendorf-Richardson (GBLOC MBFL) in the following manner:   JPPSO-NW (GBLOC JEAT) will process all shipments that are aligned with Zone II excluding Air Force and Space Force personnel.  Air Force and Space Force personnel in Zone II will be processed by JPPSO Elmendorf-Richardson (GBLOC MBFL).
- Zone III shipments will be processed by USCG Base Kodiak (GBLOC MAPS).
- Zone IV shipments will be processed by USCG Base Ketchikan (GBLOC MAPK) with the exception of Cordova which will be processed by USCG Base Kodiak (GBLOC MAPS).

**SQL** to run for zone/gbloc on addresses:
select j.code gbloc, j.name jppso, r.code rate_area, r.name rate_area_name, g.department_indicator, g.shipment_type 
from addresses a, v_locations v, re_oconus_rate_areas o, re_rate_areas r, jppso_regions j, gbloc_aors g 
where a.id = 'put_your_addr_id_here' 
and a.us_post_region_cities_id = v.uprc_id 
and v.uprc_id = o.us_post_region_cities_id 
and o.rate_area_id = r.id 
and o.id = g.oconus_rate_area_id 
and j.id = g.jppso_regions_id;

NOTE: There is an attachment in the Agility BL that has Alaska Cities by Zone to use also.
